administrator/components/com_config/src/Model/ApplicationModel.php

```diff
@@ -585,11 +585,11 @@ public function save($data)
         }
 
         // Give a warning if the cache-folder can not be opened
-        if ($data['caching'] > 0 && $data['cache_handler'] == 'file' && @opendir($path) == false) {
+        if ($data['caching'] > 0 && $data['cache_handler'] == 'file' && !@opendir($path)) {
```

Comment:
Use strict comparison with false to make it clear that the directory could not be opened
```suggestion
        if ($data['caching'] > 0 && $data['cache_handler'] == 'file' && @opendir($path) === false) {
```